### DISCUSSION TITLE
Defining Relationships Between Customer and Sales Rep Objects

### DISCUSSION THREAD

**Discussion by arulselvan008** [pimcore/pimcore#17532] [Q&A / Support]
Hi Team,

We are currently working on defining the relationship between two objects, Customer and Sales Reps, and I wanted to get your input on the best approach for this mapping.

Here are the scenarios we are dealing with:

1. Each Sales Rep can have multiple Customers mapped to them.
2. Each Customer should be mapped to only one Sales Rep and should not be allowed to have more than one Sales Rep.
Please share your thoughts on the most suitable relationship structure to implement these requirements.

Looking forward to your feedback. @kingjia90 @fashxp @NiklasBr @dvesh3 

Best regards,
Arul

**Comment by arulselvan008** [comment]
Hi,

@kingjia90 @fashxp @NiklasBr @dvesh3 anybody Can you help us?

**Reply by NiklasBr** [reply]
If `Customer` and `Sales` are both DataObjects then it can be achieved like this:

1. On `Customer` create a "Assigned Sales Representative" many-to-one relation field, limit it to the `Sales` object class.
2. On `Sales` create a "Assigned Customers" reverse object relation field, connect it to the `Customer` field above.

**Comment by arulselvan008** [comment]
@NiklasBr 
Thanks for your support and prompt response. We have achieved the relationships between customers and sales reps.

We have one more question, please see below:

We have created the PriceGroup data object for PriceListProducts (Advanced Many-to-Many) and added products. In the Advanced Many-to-Many field, we have created two editable columns: **List Price and Bucket Quantity**. Users can manually enter the list price and bucket quantity for manual orders.

**Validation:** How can we set minimum and maximum validation logic for the List Price and Bucket Quantity fields to ensure that users enter values within the allowed range? The user input values are stored in the **object_metadata_PriceGroup** table.

**Data Import:** We have existing master data for **List Price and Bucket Quantity**. Is it possible to upload this data to PriceListProducts in the PriceGroup data object or Products data object using the **data importer process**? If so, how can we implement this?

PriceGroup Class: 
![image](https://github.com/user-attachments/assets/a0bc9bf4-717e-4b74-8339-283cc0f4ce4d)

![image](https://github.com/user-attachments/assets/fbe0ebb1-6cc9-44c5-bc98-67a4f01076e0)

Products Class:
![image](https://github.com/user-attachments/assets/c837dd41-8094-42ec-b3e4-fbe8d7c97aa9)

Sample data:
![image](https://github.com/user-attachments/assets/836891ec-fc86-4f41-9810-8fc48f608752)

Need to validate for these fields and is it possible to update the date using the data importer? Looking forward to your feedback. @kingjia90 @NiklasBr

**Reply by arulselvan008** [reply]
Hi,

@kingjia90 @fashxp @NiklasBr @dvesh3 anybody Can you help us?

**Reply by NiklasBr** [reply]
Please stop tagging people this way.

**Reply by arulselvan008** [reply]
My bad. I’ll avoid this going forward. Could you please guide us on the best way to reach out to the Pimcore team for assistance with our queries? We appreciate your support!

**Reply by arulselvan008** [reply]
We have sent an email to your enterprise-support@pimcore.atlassian.net. How long should we expect to wait for a response?

**Reply by ghost** [reply]
@arulselvan008 Sorry for the late reply from us at Pimcore. Please note that we only look at the discussions going on on an irregular base as we consider them foremost a community means of helping each other. Also, as we have only limited resources, we have to focus more on our roadmap. So we cannot guarantee to answer here or even solve the questions at all.
If you have an Enterprise license and found a bug, please open a ticket in our Enterprise Issue Portal [here](https://pimcore.atlassian.net/servicedesk/customer/portal/2).
If your request does not refer to an issue in Pimcore but rather to consulting services needed, we could gladly offer this to you. In case you are interested, please contact your Partner Manager or our Sales Department: [mailto:sales@pimcore.com]. We will be happy to help you.
Thank you very much for your understanding & kind regards!
